migrations/20230811182109_auto-replies-autoassignment.js

```diff
@@ -0,0 +1,234 @@
+        )
+      )
+    );
+
+    create or replace view assignable_needs_reply as (
+      select acc.id, acc.campaign_id, acc.message_status
+      from assignable_campaign_contacts as acc
+      join campaign on campaign.id = acc.campaign_id
+      where message_status = 'needsResponse'
+      and (
+        ( 
+          acc.contact_timezone is null
+          and extract(hour from CURRENT_TIMESTAMP at time zone campaign.timezone) < campaign.texting_hours_end
+          and extract(hour from CURRENT_TIMESTAMP at time zone campaign.timezone) >= campaign.texting_hours_start
+        )
+        or
+        ( 
+          campaign.texting_hours_end > extract(hour from (CURRENT_TIMESTAMP at time zone acc.contact_timezone) + interval '10 minutes')
+          and campaign.texting_hours_start <= extract(hour from (CURRENT_TIMESTAMP at time zone acc.contact_timezone))
+        )
+      )
+    );
+
+    create or replace view assignable_needs_reply_with_escalation_tags as (
+      select acc.id, acc.campaign_id, acc.message_status, acc.applied_escalation_tags
+      from assignable_campaign_contacts_with_escalation_tags as acc
+      join campaign on campaign.id = acc.campaign_id
+      where message_status = 'needsResponse'
+      and (
+        ( 
+          acc.contact_timezone is null
+          and extract(hour from CURRENT_TIMESTAMP at time zone campaign.timezone) < campaign.texting_hours_end
+          and extract(hour from CURRENT_TIMESTAMP at time zone campaign.timezone) >= campaign.texting_hours_start
+        )
+        or
+        ( 
+          campaign.texting_hours_end > extract(hour from (CURRENT_TIMESTAMP at time zone acc.contact_timezone) + interval '10 minutes')
+          and campaign.texting_hours_start <= extract(hour from (CURRENT_TIMESTAMP at time zone acc.contact_timezone))
+        )
+      )
+    );
+
+    create or replace view assignable_campaigns_with_needs_message as (
+      select *
+      from assignable_campaigns
+      where
+        exists (
+          select 1
+          from assignable_needs_message
+          where campaign_id = assignable_campaigns.id
+        )
+        and not exists (
+          select 1
+          from campaign
+          where campaign.id = assignable_campaigns.id
+            and now() > date_trunc('day', (due_by + interval '24 hours') at time zone campaign.timezone)
+        )
+        and autosend_status <> 'sending'
+    );
+
+    create or replace view assignable_campaigns_with_needs_reply as (
+      select *
+      from assignable_campaigns
+      where exists (
+        select 1
+        from assignable_needs_reply
+        where campaign_id = assignable_campaigns.id
+      )
+    );
+
+    drop index todos_partial_idx;
+    create index todos_partial_idx on campaign_contact (campaign_id, assignment_id, message_status, is_opted_out, auto_reply_eligible) where (archived = false);
```

Comment:
Highlighting that this adds `auto_reply_eligible` to `todos_partial_idx`. I think this is a good move because the field is functionally pretty similar to `is_opted_out` but welcome any thoughts otherwise!